s3parq/publish_parq.py

```diff
@@ -290,7 +306,16 @@ def _parquet_schema(dataframe: pd.DataFrame) -> pa.Schema:
     for col, dtype in dataframe.dtypes.items():
         dtype = dtype.name
         if dtype == 'object':
-            pa_type = pa.string()
+            if custom_redshift_columns:
+                if "[Decimal(" in str(dataframe[col].values)[:9]:
+                    s = custom_redshift_columns[col]
+                    precision = int(s[s.find('DECIMAL(')+len('DECIMAL('):s.rfind(',')].strip())
+                    scale = int(s[s.find(',')+len(','):s.rfind(')')].strip())
+                    pa_type = pa.decimal128(precision=precision, scale=scale)
```

Comment:
Can this section get a small comment explaining what this chunk ends up doing? It's a bit dense to read

Reply:
Thanks! Comment added explaining this in detail.